lib/rust/ensogl/component/text/src/font.rs

```diff
@@ -0,0 +1,617 @@
+    }
+
+    /// Style setter. See the following docs to learn more:
+    /// https://fonts.google.com/knowledge/glossary/italic_axis
+    /// https://fonts.google.com/knowledge/glossary/slant_axis
+    pub fn set_style(&mut self, value: Style) {
+        match value {
+            Style::Normal => {
+                self.ital.value = 0_u16.into();
+                self.slnt.value = 0_u16.into();
+            }
+            Style::Italic => {
+                self.ital.value = 1_u16.into();
+                self.slnt.value = 0_u16.into();
+            }
+            Style::Oblique => {
+                self.ital.value = 0_u16.into();
+                self.slnt.value = 90_u16.into();
+            }
+        }
+    }
+}
+
+
+
+// ================
+// === Registry ===
+// ================
+
+shared! { Registry
+/// Structure keeping all fonts loaded from different sources.
+#[derive(Debug)]
+pub struct RegistryData {
+    embedded: Embedded,
+    fonts:          HashMap<Name,Font>,
+}
+
+impl {
+    pub fn load_default(&mut self) -> Font {
+        self.load(DEFAULT_FONT)
+    }
+
+    /// Load a font by name. The font can be loaded either from cache or from the embedded fonts'
+    /// registry if not used before. Returns the default font if the name is missing in both cache
+    /// and embedded font list.
+    pub fn load(&mut self, name:impl Into<Name>) -> Font {
+        let name = name.into();
+        self.try_load(&name).unwrap_or_else(|| {
+            event!(WARN, "Font '{name}' not found. Loading the default font.");
+            self.try_load(DEFAULT_FONT).expect("Default font not found.")
+        })
+    }
+
+    /// Load a font by name. The font can be loaded either from cache or from the embedded fonts'
+    /// registry if not used before. Returns [`None`] if the name is missing in both cache and
+    /// embedded font list.
+    pub fn try_load(&mut self, name:impl Into<Name>) -> Option<Font> {
+        let name = name.into();
+        event!(DEBUG, "Loading font: {:?}", name);
+        match self.fonts.entry(name.clone()) {
+            Entry::Occupied (entry) => Some(entry.get().clone_ref()),
+            Entry::Vacant   (entry) => {
+                self.embedded.definitions.get(&name).map(|definition| {
+                    let font: Font = match definition {
+                        FamilyDefinition::NonVariable(definition) => {
+                            let family = NonVariableFamily::from(definition);
+                            family.load_all_faces(&self.embedded);
+                            NonVariableFont::new(name, family).into()
+                        }
+                        FamilyDefinition::Variable(definition) => {
+                            let family = VariableFamily::from(definition);
+                            family.load_all_faces(&self.embedded);
+                            VariableFont::new(name, family).into()
+                        }
+                    };
```

Comment:
It uses `self.embedded`. I was thinking about refactoring it to `From<&family::Definition> for Font` but because of accessilng `self` here it doesnt make sense. I was also trying to refactor it as another method here, but because of unique access to self, this gets pretty complicated and we would beed to expose a lot of args to the function. The clarity will be lower:

<img width="1673" alt="image" src="https://user-images.githubusercontent.com/1623053/186470455-33125c5f-f65a-43c9-b35d-e031062456b2.png">